src/app/contexts/EventContext/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from 'react';
+import { node } from 'prop-types';
+import { useWindowEvent } from './useWindowEvent';
+import { useHandlerMap } from './useHandlerMap';
+
+export const EventContext = React.createContext({
+  useWindowEvent,
+  useClickTracker: () => {},
+});
+
+export const EventContextProvider = ({ children }) => {
+  const [handlerMap, setHandlerMap] = useState({});
+
+  const useClickTracker = (attr, handler) => {
+    useEffect(() => {
+      setHandlerMap(_map => ({
+        ..._map,
+        [attr]: [...(_map[attr] || []), handler],
+      }));
+
+      return function cleanup() {
```

Comment:
Is this on unmount of the useClickTracker function? I didnt know react was that clever, have you double checked this unmount ever runs?